pydantic/_internal/_generate_schema.py

```diff
@@ -351,6 +354,13 @@ def _typed_dict_schema(self, typed_dict_cls: Any) -> core_schema.TypedDictSchema
         validation_functions = ValidationFunctions(())
         serialization_functions = SerializationFunctions(())
 
+        obj_ref = f'{typed_dict_cls.__module__}.{typed_dict_cls.__qualname__}:{id(typed_dict_cls)}'
```

Comment:
please can we have a single function for this that's used in both places.

Reply:
see `pydantic._internal._core_utils.get_type_ref`